libr/anal/p/anal_ppc_gnu.c

```diff
@@ -9,9 +9,11 @@
 // XXX addr should be off_t for 64 love
 static int ppc_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *_bytes, int len) {
 //int arch_ppc_op(ut64 addr, const u8 *bytes, struct op_t *op)
-// TODO swap endian here??
-	char bytes[1024];
-	r_mem_copyendian ((ut8*)bytes, _bytes, 4, 1);
+	ut8 bytes[1024];
+	if (!anal->big_endian)
+		r_mem_swapendian(bytes, _bytes, 4);
+	else
+		memcpy(bytes, _bytes, 4);
```

Comment:
Plugins are in charge of this. If the librart who handles this (capstone) supports endianness then the plugin configures to do that. If not, it must do the job by hand

> On 02 May 2016, at 04:20, Damien Zammit <notifications@github.com> wrote:
> 
> In libr/anal/p/anal_ppc_gnu.c:
> 
> > @@ -9,9 +9,11 @@
> >  // XXX addr should be off_t for 64 love
> >  static int ppc_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *_bytes, int len) {
> >  //int arch_ppc_op(ut64 addr, const u8 *bytes, struct op_t *op)
> > -// TODO swap endian here??
> > -	char bytes[1024];
> > -	r_mem_copyendian ((ut8*)bytes, _bytes, 4, 1);
> > +	ut8 bytes[1024];
> > +	if (!anal->big_endian)
> > +		r_mem_swapendian(bytes, _bytes, 4);
> > +	else
> > +		memcpy(bytes, _bytes, 4);
> I agree its ugly, but then why does an analysis plugin need to swap endian at all in the first place? The asm backend already writes opcodes to a linear buffer.. I'm just trying to untangle this mess.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>